### DISCUSSION TITLE
Regolith 2.2 Release Thread

### DISCUSSION THREAD

**Discussion by kgilmer** [regolith-linux/regolith-desktop#772] [Announcements]
The first beta of Regolith 2.2 is available for testing.  As a beta, it is not recommended for usage in a production system.

Release notes (WIP): https://regolith-desktop.com/docs/reference/Releases/regolith-2.2-release-notes/

Installation instructions: Follow in the installation instructions for your install target on the website, except substitute `testing` for `release` in the `apt` line.  For example, to install into an Ubuntu 22.04 release, this is the `apt` line:

```
echo deb "[arch=amd64 signed-by=/usr/share/keyrings/regolith-archive-keyring.gpg] \
https://regolith-desktop.org/testing-ubuntu-jammy-amd64 jammy main" | \
sudo tee /etc/apt/sources.list.d/regolith.list
```

Kindly report bugs to https://github.com/regolith-linux/regolith-desktop/issues

**Comment by kgilmer** [comment]
The second beta of Regolith 2.2 is now available for testing. All the links above continue to be valid for this beta.  To get this version, follow the steps above to setup the `apt` repo.  Additionally, an ISO image of Regolith Linux 2.2 based on Ubuntu 22.10 (kinetic) is available here: https://github.com/regolith-linux/regolith-ubuntu-iso-builder/releases/tag/ubuntu-kinetic-2.2.0-20221108_091534

**Comment by kgilmer** [comment]
NOTE: when upgrading to this beta, users should execute `apt dist-upgrade`.  This allows deprecated packages to be removed and replaced with newer packages.  Beta 2 updated packaging metadata and i3 configuration to allow users to avoid the requirement of installing `regolith-control-center` and instead rely on the upstream version in the case of system dependencies that do not line up properly to Ubuntu releases.  This seems to be the case in some downstream variants of Ubuntu.  While these variants are not tested or supported, Regolith should be adaptable as possible to conform to the user's system.

**Comment by kgilmer** [comment]
Regolith 2.2 beta 3 is now available.  ISO link: https://github.com/regolith-linux/regolith-ubuntu-iso-builder/releases/download/ubuntu-kinetic-2.2.0-20221126_002525/regolith-ubuntu-kinetic-2.2.0.zip

**Comment by kgilmer** [comment]
Beta 4 brings some important fixes to `ilia` and some minor visual polish to some looks.

**Comment by jc00ke** [comment]
I saw the email for 2.2, congrats! However, I'm running into an issue upgrading:

```
➜ sudo apt update
Get:1 https://cli.github.com/packages stable InRelease [3,917 B]
Ign:2 https://regolith-desktop.org/release-ubuntu-kinetic-amd64 kinetic InRelease
Hit:3 http://us.archive.ubuntu.com/ubuntu kinetic InRelease
Get:4 http://apt.insync.io/ubuntu kinetic InRelease [5,537 B]
Hit:5 https://brave-browser-apt-release.s3.brave.com stable InRelease
Hit:6 https://downloads.1password.com/linux/debian/amd64 beta InReleases
Get:7 https://pkgs.tailscale.com/stable/ubuntu kinetic InRelease
Hit:8 https://deb.beekeeperstudio.io stable InRelease
Err:9 https://regolith-desktop.org/release-ubuntu-kinetic-amd64 kinetic Release
  404  Not Found [IP: 104.238.213.207 443]
Hit:10 http://archive.ubuntu.com/ubuntu kinetic InRelease
Hit:11 https://updates.signal.org/desktop/apt xenial InRelease
Hit:12 http://us.archive.ubuntu.com/ubuntu kinetic-updates InRelease
Hit:13 http://us.archive.ubuntu.com/ubuntu kinetic-backports InRelease
Hit:14 http://security.ubuntu.com/ubuntu kinetic-security InRelease
Hit:15 http://security.ubuntu.com/ubuntu focal-security InRelease
Hit:16 https://ppa.launchpadcontent.net/mozillateam/ppa/ubuntu kinetic InRelease
Reading package lists... Done
E: The repository 'https://regolith-desktop.org/release-ubuntu-kinetic-amd64 kinetic Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
```

**Reply by kgilmer** [reply]
Hmm, yeah something didn't work.  Investigating..

**Reply by kgilmer** [reply]
Looks like I forgot to enable the `release` stage for `kinetic` :sweat_smile:   Building in now, will update the thread when it's available.

**Reply by jc00ke** [reply]
LGTM now! :tada: 

```
➜ sudo apt update; apt list --upgradable
Get:1 https://cli.github.com/packages stable InRelease [3,917 B]
Hit:2 https://brave-browser-apt-release.s3.brave.com stable InRelease
Hit:3 https://deb.beekeeperstudio.io stable InRelease
Hit:4 https://downloads.1password.com/linux/debian/amd64 beta InRelease
Hit:5 http://us.archive.ubuntu.com/ubuntu kinetic InRelease
Hit:6 http://security.ubuntu.com/ubuntu kinetic-security InRelease
Get:7 https://pkgs.tailscale.com/stable/ubuntu kinetic InRelease
Hit:8 https://updates.signal.org/desktop/apt xenial InRelease
Hit:9 http://security.ubuntu.com/ubuntu focal-security InRelease
Hit:10 http://us.archive.ubuntu.com/ubuntu kinetic-updates InRelease
Hit:11 https://regolith-desktop.org/release-ubuntu-kinetic-amd64 kinetic InRelease
Hit:12 http://archive.ubuntu.com/ubuntu kinetic InRelease
Hit:13 http://us.archive.ubuntu.com/ubuntu kinetic-backports InRelease
Get:14 http://apt.insync.io/ubuntu kinetic InRelease [5,537 B]
Hit:15 https://ppa.launchpadcontent.net/mozillateam/ppa/ubuntu kinetic InRelease
Fetched 15.0 kB in 2s (8,203 B/s)
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
All packages are up to date.
Listing... Done
```

**Reply by jc00ke** [reply]
Well, at least the release file is there. Not sure if I've already got all the latest versions or not.

**Reply by kgilmer** [reply]
Everything should be working now, LMK if you still see problems @jc00ke :incoming_envelope:

**Reply by jc00ke** [reply]
That's working now, but I have nothing to upgrade. If I'm up to date, what version of `ilia` should I have? Anything else I can test?

**Reply by kgilmer** [reply]
`ilia` should be at `0.12.29`.  You could check the contents of `/etc/regolith/version` as well.

**Reply by jc00ke** [reply]
```
➜ ilia --version
Gtk-Message: 07:19:02.595: Failed to load module "xapp-gtk3-module"
ilia version 0.12
~
➜ cat /etc/regolith/version
REGOLITH_VERSION=2.2
```

**Comment by citosid** [comment]
Maybe this is a dumb question, but there is any place where to find the changes in 2.2? I can't find them anywhere

**Reply by kgilmer** [reply]
Not dumb at all, here you go: https://regolith-desktop.com/docs/reference/Releases/regolith-2.2-release-notes/

**Comment by cgiuliani** [comment]
The option `--class=floating_window` is no longer available?

**Reply by kgilmer** [reply]
I believe this is from GTK4, not i3 or anything else.  But yeah it's gone it seems :shrug:

**Comment by spoelstraethan** [comment]
Just some feedback on the ISO distribution.

Putting the md5sum inside the zip file with the ISO changes the md5sum of the zip (and typically users should care about the checksum of the _download_ to ensure it wasn't tampered or corrupted in transit, assuming that just because it unzips and the ISO's checksum matches the md5 file doesn't take into account that a malicious person could rewrite the zip contents on the fly and modify the checksum file and ISO or more easily just substitute an alternate zip file since the zip archive doesn't have a published checksum.

In addition the extra file inside the zip makes it incompatible with the Chrome Recovery Utility extension which is a great alternative to Etcher or Rufus for Chromebooks, it can transparently "mount" a zip and write the .img/.bin/.iso inside to a USB, and if you extracted the Regolith zip since the Recovery Tool only acknowledges `.bin` or `.zip` files in the file browser so you have to append `.bin` to the name via the `Files` app so it shows up. I confirmed this by extracting only the ISO from the zip, deleting the zip, then creating a new zip in the ChromeOS Files app and selecting it as the "Local Image" in the recovery tool and it wrote it to USB the same as it did when I appended .bin to the .iso extension.
